Source/Evolve/WarpXEvolve.cpp

```diff
@@ -523,6 +517,14 @@ void WarpX::HandleParticlesAtBoundaries (int step, amrex::Real cur_time, int num
         }
     }
 
+    // interact the particles with EB walls (if present)
+#ifdef AMREX_USE_EB
+    mypc->ScrapeParticles(amrex::GetVecOfConstPtrs(m_distance_to_eb));
+    m_particle_boundary_buffer->gatherParticlesFromEmbeddedBoundaries(*mypc, amrex::GetVecOfConstPtrs(m_distance_to_eb));
+#endif
+
+    mypc->deleteInvalidParticles();
```

Comment:
This should be inside the EB block above since it is only needed there. The Redistribute above will have already deleted invalid particles outside of the domain.